File: TIENDARENEGUL/src/Producto.java
import java.util.Date;

// Clase Producto que servirá como clase base para los diferentes tipos de productos
class Producto {
    private String nombre;
    private double precio;
    private String fechaImportacion;
    private String numeroSerie;
    private int stock;

    // Constructor
    public Producto(String nombre, double precio, String fechaImportacion, String numeroSerie, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.fechaImportacion = fechaImportacion;
        this.numeroSerie = numeroSerie;
        this.stock = stock;
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public String getFechaImportacion() {
        return fechaImportacion;
    }

    public void setFechaImportacion(String fechaImportacion) {
        this.fechaImportacion = fechaImportacion;
    }

    public String getNumeroSerie() {
        return numeroSerie;
    }

    public void setNumeroSerie(String numeroSerie) {
        this.numeroSerie = numeroSerie;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    // Método para agregar al stock
    public void agregarStock(int cantidad) {
        if (cantidad > 0) {
            this.stock += cantidad;
            System.out.println(cantidad + " " + this.nombre + " agregadas al stock.");
        } else {
            System.out.println("La cantidad debe ser mayor que cero.");
        }
    }

    // Método para eliminar del stock
    public void eliminarStock(int cantidad) {
        if (cantidad > 0 && cantidad <= this.stock) {
            this.stock -= cantidad;
            System.out.println(cantidad + " " + this.nombre + " eliminadas del stock.");
        } else {
            System.out.println("La cantidad no es válida.");
        }
    }

}

File: TIENDARENEGULPY/Producto.py
class Producto:
    def __init__(self, nombre, precio, fecha_importacion, numero_serie, stock):
        self.nombre = nombre
        self.precio = precio
        self.fecha_importacion = fecha_importacion
        self.numero_serie = numero_serie
        self.stock = stock

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_precio(self):
        return self.precio

    def set_precio(self, precio):
        self.precio = precio

    def get_fecha_importacion(self):
        return self.fecha_importacion

    def set_fecha_importacion(self, fecha_importacion):
        self.fecha_importacion = fecha_importacion

    def get_numero_serie(self):
        return self.numero_serie

    def set_numero_serie(self, numero_serie):
        self.numero_serie = numero_serie

    def get_stock(self):
        return self.stock

    def set_stock(self, stock):
        self.stock = stock

    def agregar_stock(self, cantidad):
        if cantidad > 0:
            self.stock += cantidad
            print(str(cantidad) + " " + self.nombre + " agregadas al stock.")
        else:
            print("La cantidad debe ser mayor que cero.")

    def eliminar_stock(self, cantidad):
        if 0 < cantidad <= self.stock:
            self.stock -= cantidad
            print(str(cantidad) + " " + self.nombre + " eliminadas del stock.")
        else:
            print("La cantidad no es válida.")
